lib/ansible/galaxy/dependency_resolution/dataclasses.py

```diff
@@ -323,14 +352,24 @@ def from_requirement_dict(cls, collection_req, art_mgr, validate_signature_optio
                 req_signature_sources = [req_signature_sources]
             req_signature_sources = frozenset(req_signature_sources)
 
+        if (
+            req_version and req_type == 'subdirs'
+        ):
+            raise AnsibleError(
+                f"The {req_name} 'version' key is not applicable when 'type' is set to 'subdirs'."
```

Comment:
I think this is an internal type that is not even exposed to the end-users. So showing it like this would probably be confusing.